apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1539,6 +1540,58 @@ static void tc_pthread_pthread_setcanceltype(void)
 }
 #endif
 
+/**
+* @fn                   :tc_pthread_pthread_mutex_consistent
+* @brief                :This tc tests pthread_mutex_consistent()
+* @Scenario             :the pthread_mutex_consistent() function can be used to mark the state protected by the mutex referenced by mutex as consistent again
+*                        and this situtaion arises, If an owner of a robust mutex terminates while holding the mutex,
+*                        the mutex becomes inconsistent and the next thread that acquires the mutex lock shall be notified of the state.
+*                        In this case, the mutex does not become normally usable again until the state is marked consistent.
+* @API'scovered         :pthread_setcanceltype
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void *test_mutex_consistent(void *param)
+{
+	int ret = -1;
+	ret = pthread_mutex_lock(&g_mutex);
+	if (ret != 0) {
+		printf("pthread_mutexlock was not successful, errno = %d", ret);
+	}
+}
+
+static void tc_pthread_pthread_mutex_consistent(void)
+{
+	int ret;
+	pthread_t thread1;
+	pthread_t thread2;
+
+	pthread_mutexattr_init(&g_mutexattr);
+	pthread_mutexattr_setrobust(&g_mutexattr, PTHREAD_MUTEX_ROBUST);
+	pthread_mutex_init(&g_mutex, &g_mutexattr);
+
+	ret = pthread_create(&thread1, NULL, &test_mutex_consistent, NULL);
+	pthread_join(thread1, NULL);
+	TC_ASSERT_EQ("pthread_mutex_consistent", ret, 0);
+
+
+	ret = pthread_kill(thread1, 0);
+	if (ret != 0) {
+		printf("Thread 1 has died.\n");
+	}
+
+	pthread_mutex_consistent(&g_mutex);
+
+	ret = pthread_create(&thread2, NULL, &test_mutex_consistent, NULL);
```

Comment:
I think we'd better make two pthreads independently.
1st is locking the mutex and finished,
2nd is trying to lock the previous mutex.
And check that return of `pthread_mutex_lock` is `EOWNERDEAD` or not from 2nd pthread.

Reply:
Okay. How about adding 2nd testcase scenario also?
Because, it is special to return `EOWNERDEAD` from `pthread_mutex_lock()`, so it is much better testcase for testing inconsistent mutex.

Reply:
sorry, there is no first case. today I checked so TC was failing but yesterday it was pasing.
so TC is for only 2nd case.